diff --git a/third_party/dxc/cmake/modules/CrossCompile.cmake b/third_party/dxc/cmake/modules/CrossCompile.cmake
--- a/third_party/dxc/cmake/modules/CrossCompile.cmake
+++ b/third_party/dxc/cmake/modules/CrossCompile.cmake
@@ -19,6 +19,7 @@
   add_custom_command(OUTPUT ${LLVM_${target_name}_BUILD}/CMakeCache.txt
     COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"
-        ${CROSS_TOOLCHAIN_FLAGS_${target_name}} ${CMAKE_SOURCE_DIR}
+        -DCMAKE_BUILD_TYPE=Release -DLLVM_INCLUDE_TESTS=OFF -DHLSL_INCLUDE_TESTS=OFF -DLLVM_TARGETS_TO_BUILD=
+        ${CROSS_TOOLCHAIN_FLAGS_${target_name}} ${LLVM_MAIN_SRC_DIR}
     WORKING_DIRECTORY ${LLVM_${target_name}_BUILD}
     DEPENDS ${LLVM_${target_name}_BUILD}
     COMMENT "Configuring ${target_name} LLVM...")
diff --git a/third_party/dxc/cmake/modules/TableGen.cmake b/third_party/dxc/cmake/modules/TableGen.cmake
--- a/third_party/dxc/cmake/modules/TableGen.cmake
+++ b/third_party/dxc/cmake/modules/TableGen.cmake
@@ -101,7 +101,7 @@ macro(add_tablegen target project)
       set(${project}_TABLEGEN_EXE ${${project}_TABLEGEN_EXE} PARENT_SCOPE)
 
       add_custom_command(OUTPUT ${${project}_TABLEGEN_EXE}
-        COMMAND ${CMAKE_COMMAND} --build . --target ${target} --config Release
+        COMMAND ${CMAKE_COMMAND} --build . --target ${target} --config Release --parallel 1
         DEPENDS CONFIGURE_LLVM_NATIVE ${target}
         WORKING_DIRECTORY ${LLVM_NATIVE_BUILD}
         COMMENT "Building native TableGen...")
 
